main.tf

```diff
@@ -163,16 +163,16 @@ resource "aws_cloudwatch_log_group" "log_group" {
 }
 
 resource "aws_sns_topic_subscription" "sns_topic_arn" {
-  topic_arn = var.sns_topic_arn
+  count     = var.provision_trigger ? length(var.sns_topic_arn) : 0
+  topic_arn = var.sns_topic_arn[count.index]
   protocol  = "lambda"
   endpoint  = aws_lambda_function.logs_to_datadog.arn
 }
 
 resource "aws_lambda_permission" "sns_topic_arn" {
-  statement_id  = "AllowExecutionFromSNS"
```

Comment:
any reason why we're getting rid of the statement id attribute? does it need to be unique if there are more than one?

Reply:
When you are creating more than on trigger, it fails on that statement when it creates the second one. In what I read, it shouldn't be necessary. It has worked without it in my testing. We can figure a way to put it back if it becomes a problem in the future.